src/lib-components/overlays/Popover/Popover.vue

```diff
@@ -1,229 +1,59 @@
-<script lang="ts">
-export type PopoverPosition =
-  | "top-left"
-  | "top-center"
-  | "top-right"
-  | "bottom-left"
-  | "bottom-center"
-  | "bottom-right"
-  | "left"
-  | "right"
-  | "auto"
-  | "none"
-</script>
 <script lang="ts" setup>
-import { throttle } from "@/helpers/Throttle"
 import {
   Popover as HeadlessPopover,
   PopoverButton as HeadlessPopoverButton,
   PopoverPanel as HeadlessPopoverPanel,
 } from "@headlessui/vue"
-import { computed, onMounted, onUnmounted, ref } from "vue"
+import { computed, ref } from "vue"
+import {
+  useFloating,
+  offset,
+  autoPlacement,
+  autoUpdate,
+  shift,
+} from "@floating-ui/vue"
+import type { Placement } from "@floating-ui/vue"
 
 const props = withDefaults(
   defineProps<{
     as?: string
-    position?: PopoverPosition
+    position?: Placement
   }>(),
   {
     as: "div",
-    position: "auto",
```

Comment:
I think the auto string felt more explicit. Maybe Im just used to it at this point, but undefined is not clear at first grok that it is then using auto.

Reply:
Yeah, that's a great point from an API clarity standpoint. Updated!